### PR TITLE
adding ABNF section

### PR DESCRIPTION
**Pull request by jscancella**
This Pull Request integrates #23 into the index.html.

### CODE REVIEW

**Comment by jscancella** [comment]
@ruebot sure!
ABNF is often used to auto generate code by many projects. This would those projects to automatically create code/software that integrates with the specification. Additionally, many RFC specifications like to have ABNF as it defines in a very concrete way a specification.

I think it is also just another way to communicate the specification and make sure there is no ambiguity in the specification that can cause problems later (like what happened in the BagIt specification).

File: index.html [hunk 1]
```diff
@@ -382,5 +382,78 @@ <h3>
...
+        profile-info                = DQUOTE "BagIt-Profile-Info" DQUOTE ":{" profile-identifier profile-bagit-version profile-organization-name profile-name profile-email profile-description profile-version "},"
+        profile-identifier          = DQUOTE "BagIt-Profile-Identifier" DQUOTE ":" DQUOTE absolute-URI DQUOTE ","
+        profile-bagit-version       = DQUOTE "BagIt-Profile-Version" DQUOTE ":" version ","
+        profile-organization-name   = DQUOTE "Source-Organization" DQUOTE ":" value ","
+        profile-name                = DQUOTE "Contact-Name" DQUOTE ":" value ","
+        profile-email               = DQUOTE "Contact-Email" DQUOTE ":" email-address ","
+        email-address               = DQUOTE addr-spec DQUOTE
+        profile-description         = DQUOTE "External-Description" DQUOTE ":" value ","
+        profile-version             = DQUOTE "Version" DQUOTE ":" version ","
+        version                     = DQUOTE 1*DIGIT "." 1*DIGIT DQUOTE
```
Comment (tdilauro) [inline index.html]: `version` rule supports multiple other rules with varying numbers of dots in their version numbers. A more generalized version production would be. Might considering calling it "dotted-version" to allow for more generalized (undotted) version strings 
```
version = DQUOTE 1*DIGIT *("." 1*DIGIT) DQUOTE

**Comment by jscancella** [comment]
@tdilauro I will also be gone most of January and unable to work on this. 
* Maintainability/GROK - if you have suggestions I am open to looking at them and incorporating them into the pull request
* correctness - I followed the order that was given in the normative section. I didn't see it say anywhere that there was a specific order they had to be in, and when parsing JSON you shouldn't need a particular order
* precision 
 * VCHAR - unless you really want to get ugly ABNF rules, I don't see how you are going to specify that you have to escape the quotes in values.
 * case sensitive - good to know it is by default case INSENSITIVE. However, do we really want to force case senstivity for what are basically map keys? @ruebot, what are your thoughts on this?

**Comment by tdilauro** [comment]
@jscancella Ugh. I just wanted to get my vague concerns out there, but didn't mean to cause any undo stress. I'll try to have a look sooner rather than later, but have to balance family vacation time. If it doesn't work out, we can always iterate later.

> @tdilauro I will also be gone most of January and unable to work on this until I get back. Hopefully I can finish it before I leave.

**Comment by jscancella** [comment]
> @jscancella Ugh. I just wanted to get my vague concerns out there, but didn't mean to cause any undo stress. I'll try to have a look sooner rather than later, but have to balance family vacation time. If it doesn't work out, we can always iterate later.

Don't worry about it, family is more important especially this time of year. Look at it when you get a chance, but don't stress.

**Comment by jscancella** [comment]
Maybe it didn't make it into the specification, but I remember talking about https://tools.ietf.org/html/rfc8493#section-6.1.1.1 with Chris Adams and how we should try and preserve case, but not force anyone to be case sensitive when looking up information.
For instance, if we have a `bag` object and want to get a particular key in the `bag-info` we shouldn't be case sensitive. But when we go to write the `bag-info` (or other parts of the bag) we should preserve the case sensitivity entered and write it exactly the same out to file.

I looked at https://tools.ietf.org/html/rfc7159 but couldn't find where it says that `strings` are case sensitive. Only that they are required to test for equality (see [section 8.3](https://tools.ietf.org/html/rfc7159#section-8.3)).

**Comment by jscancella** [comment]
Also I just found in [ECMA-404](http://www.ecma-international.org/publications/files/ECMA-ST/ECMA-404.pdf) 

> An object structure is represented as a pair of curly bracket tokens surrounding zero or more name/value pairs. A name is a string. A single colon token follows each name, separating the name from the value. A single comma token separates a value from a following name. The JSON syntax does not impose any restrictions on the strings used as names, does not require that name strings be unique, and does not assign any significance to the ordering of name/value pairs. These are all semantic considerations that may be defined by JSON processors or in specifications defining specific uses of JSON for data interchange.

Perhaps I am not understanding, but from the wording above it seems it is up to the individual parsers to determine if the `name` is case sensitive.

**Comment by tdilauro** [comment]
I think that's fine, but we should be explicit about it. Additionally, the [current release](https://github.com/bagit-profiles/bagit-profiles-validator/releases/tag/1.3.0) of the validator and [my recently committed PR](https://github.com/bagit-profiles/bagit-profiles-validator/commit/7246a73e8538b62f6edd31f789e06e4ca011a70b) are case sensitive.

File: index.html [hunk 2]
```diff
@@ -58,7 +58,7 @@ <h1>Introduction</h1>
       <p>
         <pre>
 
-	                        BagIt Profile JSON file
```
Comment (jscancella) [inline index.html]: this is from merging latest from master

**Comment by jscancella** [comment]
Unless I am misunderstanding I think it is resolved - make it case insensitive. I haven't seen any proposed updates on the forum for this either, so should we go ahead and merge it?

**Comment by tdilauro** [comment]
The [latest validator](https://github.com/bagit-profiles/bagit-profiles-validator/commit/41e7eecf2d1c71effe39911fbc723fb1e74c0af0) is case sensitive. Shouldn't these be in sync?

**Pull request merged**